src/main/java/com/example/animalshelter/service/AnimalServiceImpl.java

```diff
@@ -20,14 +19,18 @@ public AnimalServiceImpl(AnimalRepository animalRepository) {
 
     @Override
     public Animal add(Animal animal) {
-        if (!animalRepository.add(animal)) {
-            throw new AddAnimalException(REPOSITORY_EXCEPTION_MSG);
-        }
-        return animal;
+       return animalRepository.save(animal);
     }
 
     @Override
     public Animal delete(Long id) {
-        return animalRepository.delete(id);
+        Optional<Animal> animalToDelete = animalRepository.findById(id);
+
+        if(animalToDelete.isEmpty()){
+           throw new AnimalNotFoundException(id);
+        }
+
+        animalRepository.delete(animalToDelete.get());
+        return animalToDelete.get();
```

Comment:
I would think here about `Derived Delete Query` (returning id of modified resource is totally enough) or modify this code using optional functions.

```suggestion
       Animal animalToDelete = animalRepository.findById(id)
           .orElseThrow(() -> new AnimalNotFoundException(id));
        animalRepository.delete(animalToDelete);
        return animalToDelete;
```

Reply:
I have changed and now we return id of deleted animal as suggested.